Tunesviewer keeps crashing on newer Android versions.

For example, trying to download something triggers an error on Pixel 3 API 28:
```
2019-09-24 21:43:15.401 19813-19813/com.tunes.viewer E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.tunes.viewer, PID: 19813
    java.lang.RuntimeException: Unable to start service com.tunes.viewer.FileDownload.DownloadService@1287e53 with Intent { cmp=com.tunes.viewer/.FileDownload.DownloadService (has extras) }: java.lang.IllegalArgumentException: Reserved id
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3686)
        at android.app.ActivityThread.access$1600(ActivityThread.java:199)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1681)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalArgumentException: Reserved id
        at android.os.Parcel.createException(Parcel.java:1946)
        at android.os.Parcel.readException(Parcel.java:1910)
        at android.os.Parcel.readException(Parcel.java:1860)
        at android.app.INotificationManager$Stub$Proxy.createNotificationChannels(INotificationManager.java:1751)
        at android.app.NotificationManager.createNotificationChannels(NotificationManager.java:541)
        at android.app.NotificationManager.createNotificationChannel(NotificationManager.java:529)
        at com.tunes.viewer.FileDownload.Notifier.makeNotification(Notifier.java:97)
        at com.tunes.viewer.FileDownload.Notifier.<init>(Notifier.java:39)
        at com.tunes.viewer.FileDownload.DownloaderTask.onPreExecute(DownloaderTask.java:352)
        at android.os.AsyncTask.executeOnExecutor(AsyncTask.java:648)
        at android.os.AsyncTask.execute(AsyncTask.java:595)
        at com.tunes.viewer.FileDownload.DownloadService.onStart(DownloadService.java:117)
        at android.app.Service.onStartCommand(Service.java:497)
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3667)
        at android.app.ActivityThread.access$1600(ActivityThread.java:199) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1681) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
     Caused by: android.os.RemoteException: Remote stack trace:
        at com.android.server.notification.RankingHelper.createNotificationChannel(RankingHelper.java:616)
        at com.android.server.notification.NotificationManagerService$10.createNotificationChannelsImpl(NotificationManagerService.java:2227)
        at com.android.server.notification.NotificationManagerService$10.createNotificationChannels(NotificationManagerService.java:2242)
        at android.app.INotificationManager$Stub.onTransact(INotificationManager.java:256)
        at android.os.Binder.execTransact(Binder.java:731)
```

--- a/.idea/gradle.xml
+++ b/.idea/gradle.xml
@@ -3,12 +3,15 @@
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
-        <compositeConfiguration>
-          <compositeBuild compositeDefinitionSource="SCRIPT" />
-        </compositeConfiguration>
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
+        <option name="modules">
+          <set>
+            <option value="$PROJECT_DIR$" />
+          </set>
+        </option>
         <option name="resolveModulePerSourceSet" value="false" />
+        <option name="testRunner" value="PLATFORM" />
       </GradleProjectSettings>
     </option>
   </component>

--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
       package="com.tunes.viewer"
-      android:versionCode="17" android:versionName="1.3.0">
+      android:versionCode="18" android:versionName="1.3.1">
       <!-- Android tablet compatibility -->
 
     <uses-permission android:name="android.permission.INTERNET"/>
@@ -19,7 +19,7 @@
         </service> -->
 
         <provider
-            android:name="android.support.v4.content.FileProvider"
+            android:name="androidx.core.content.FileProvider"
             android:authorities="${applicationId}.provider"
             android:exported="false"
             android:grantUriPermissions="true">

--- a/build.gradle
+++ b/build.gradle
@@ -14,7 +14,7 @@ buildscript {
      * of the Gradle plugin for the buildscript.
      */
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.1'
+        classpath 'com.android.tools.build:gradle:3.5.0'
 
 
 
@@ -39,9 +39,9 @@ apply plugin: 'com.android.application'
  * following line includes all the JAR files in the libs directory.
  */
 dependencies {
-    compile fileTree(dir: 'libs', include: ['*.jar'])
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
 
-    implementation "com.android.support:support-core-utils:26.1.0"
+    implementation 'androidx.legacy:legacy-support-core-utils:1.0.0'
     // Add other library dependencies here (see the next step)
 }
 
@@ -51,7 +51,7 @@ dependencies {
  */
 android {
     useLibrary  'org.apache.http.legacy'
-    compileSdkVersion 26
+    compileSdkVersion 29
     buildToolsVersion "28.0.3"
 
     /**
@@ -86,6 +86,6 @@ android {
     }
     defaultConfig {
         minSdkVersion 14
-        targetSdkVersion 26
+        targetSdkVersion 29
     }
 }

--- a/src/com/tunes/viewer/FileDownload/Notifier.java
+++ b/src/com/tunes/viewer/FileDownload/Notifier.java
@@ -90,7 +90,8 @@ public class Notifier {
 		if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O)
 		{
 			int importance = NotificationManager.IMPORTANCE_HIGH;
-			NotificationChannel notificationChannel = new NotificationChannel(NotificationChannel.DEFAULT_CHANNEL_ID, "NOTIFICATION_CHANNEL_NAME", importance);
+			String notificationChannelId = "tunesviewer_app_notifications";
+			NotificationChannel notificationChannel = new NotificationChannel(notificationChannelId, "NOTIFICATION_CHANNEL_NAME", importance);
 			notificationChannel.enableLights(false);
 			notificationChannel.enableVibration(false);
 			//mBuilder.setChannelId(NOTIFICATION_CHANNEL_ID);

--- a/src/com/tunes/viewer/WebView/JSInterface.java
+++ b/src/com/tunes/viewer/WebView/JSInterface.java
@@ -31,7 +31,7 @@ import android.content.Context;
 import android.content.DialogInterface;
 import android.content.Intent;
 import android.net.Uri;
-import android.support.v4.content.FileProvider;
+import androidx.core.content.FileProvider;
 import android.text.ClipboardManager;
 import android.util.Log;
 import android.widget.TextView;

